src/datamodel/validator/UUri.cpp

```diff
@@ -109,7 +109,15 @@ ValidationResult isValidDefaultSource(const v1::UUri& uuri) {
 		return {false, Reason::EMPTY};
 	}
 
-	return isValidRpcResponse(uuri);
+	if (uses_wildcards(uuri)) {
+		return {false, Reason::DISALLOWED_WILDCARD};
+	}
+
+	if (uuri.resource_id() == 0) {
+		return {false, Reason::BAD_RESOURCE_ID};
+	}
```

Comment:
Using `0` as the default source's resource ID was intentional - it is unset and needs to be set when used (i.e. should fail validation when used in a communication module).

I've actually been considering adding an interface to the L2 communication classes that _only_ takes resource ID instead of the whole source URI to make this clearer to the user of the interface. Something like:

```
	NotificationSource(std::shared_ptr<transport::UTransport> transport,
	                   uint16_t source_resource_id, v1::UUri&& sink,
	                   std::optional<v1::UPayloadFormat> payload_format = {},
	                   std::optional<v1::UPriority> priority = {},
	                   std::optional<std::chrono::milliseconds> ttl = {}) {

		auto source_uri = transport->getDefaultSource();
		source_uri.set_resource_id(source_resource_id);
		...
	}
```

We would still need to make sure the URIs are being validated correctly in the message builder, but I think this would make the expectation more clear at the communication layer.

Reply:
So if it was intentionally set to zero, should the UUri validator validate that its zero and nothing else?  IE Change the check to != 0 ?  Or are we not wanting to be so deliberate?